models/src/main/java/com/vimeo/networking2/DashVideoFile.kt

```diff
@@ -0,0 +1,52 @@
+package com.vimeo.networking2
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+import com.vimeo.networking2.annotations.Internal
+import com.vimeo.networking2.common.VideoFile
+import java.io.Serializable
+import java.util.*
+
+/**
+ * A video file that represents a DASH stream.
+ */
+@JsonClass(generateAdapter = true)
+data class DashVideoFile(
```

Comment:
One of the design principles of V2 is that we are not using inheritance if possible. The design of these classes was completely changed, as it was not correctly designed. The `quality` property existed on the `HlsVideoFile` and the `DashVideoFile` even though it was unnecessary (the quality was always `HLS` or `DASH`). If you look at the way the files were used, the `VideoFile` class was used as an interface to accept HLS, DASH, or progressive streams. Adding `DASH` and `HLS` as artificial qualities was just a hacky way to differentiate between stream types. The use of inheritance also obfuscated common properties when viewing the class. Since the type system already has a built-in way of determining whether it's an HLS or DASH stream, by simply using `instanceof` in java and `is` in kotlin, I simplified the designs of these classes and cleaned them up. They still share the common `VideoFile` interface, where the properties are defined. 

The only other common property shared between `DashVideoFile` and `HlsVideoFile` is the `live` property, which was originally shared in the `VideoFile` abstract class. However, this property does not exist on the `ProgressiveVideoFile` DTO, since live streams are always DASH or HLS, so I opted to duplicate it across the two files, as I don't think it warrants creating a separate interface.